app/Models/Achievement.php

```diff
@@ -282,6 +282,18 @@ public function getPointsWeightedAttribute(): int
 
     // == mutators
 
+    public function promote(): void
+    {
+        $this->Flags = AchievementFlag::OfficialCore;
+        $this->save();
+    }
+
+    public function demote(): void
+    {
+        $this->Flags = AchievementFlag::Unofficial;
+        $this->save();
+    }
```

Comment:
I see these were added for testing. I'd recommend against having them be part of the official model, as it implies that's all that needs to happen when an achievement is promoted/demoted. In reality, the player and game metrics need to be updated as well.

Reply:
That's a fair point I hadn't considered. Removed.